server: remove stale comment

tamird opened the pull request:
- util.ListenAndServe and the inline comments are gone as of 41b83f8
- the TLS and non-TLS cases are identical since 7017b82
- Go supports h2c now (https://godoc.org/golang.org/x/net/http2/h2c)

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47586)
<!-- Reviewable:end -->

tamird committed:
server: remove stale comment

- util.ListenAndServe and the inline comments are gone as of 41b83f8
- the TLS and non-TLS cases are identical since 7017b82
- Go supports h2c now (https://godoc.org/golang.org/x/net/http2/h2c)

Release note: None

benesch commented:
Running out of things to do in quarantine? 😄

jordanlewis commented:
👋 what happened to your 🦷 🖌️ photo?!

andreimatei commented:
Yeah we've been discussing the photo

tbg commented:
bors r+

Thanks Tamir!

tamird commented:
@jordanlewis I got you covered https://twitter.com/tamird

jordanlewis commented:
![image](https://user-images.githubusercontent.com/43821/79592666-4d6be880-80a8-11ea-8732-8bd61721fb64.png)

jordanlewis commented:
![image](https://media.giphy.com/media/kaeaX9qTd78enHxL9Y/giphy.gif)
